### DISCUSSION TITLE
GazeboDrone and External API/Sensors

### DISCUSSION THREAD

**Discussion by nikitabeebe** [microsoft/AirSim#3954] [Support Q&A] [answered]
I know the GazeboDrone project prevents other kinematic functions, like setting position, ect. But does it disable the use of IMU, GPS, and Image sensors as well?

I have a script I'm trying to run to log the IMU, GPS, and Image data to a .log file (running separately in a python script while running GazeboDrone with the build command). Should all the sensors still run as expected despite using a external physics engine?

**Comment by jonyMarino** [comment, marked as answer]
Hi @nikitabeebe! currently, GazeboDrone updates only the drone pose, so you won't get IMU readings from AirSim. You should be able to get GPS location and image data correctly. 

Can you get IMU readings from Gazebo to accomplish what you are needing?

**Reply by nikitabeebe** [reply]
Yes probably. I had tried to get the GPS data, but maybe I'm not calling it correctly? If I call 'client.getGpsData()' along with gnss.velocity or other gnss commands, it either returns zero through out the flight, or an impossible value (like -400000000 for velocity, which I'm guessing is a min. threshold), but never changing despite my flight path - which is rendering in both gazebo and UE4, so no problems visually.

Is there another command or proper way to call the GPS data? I mainly need altitude, latitude, and longitude, with velocity as a plus.

**Reply by jonyMarino** [reply]
You can also get the coordinates through getVehiclePose and then translate that to altitude, latitude, and longitude.
But I think it would be better if you get position and velocity from the Gazebo simulator itself.

**Reply by aminnamin** [reply]
hi how can i translate the coordinate to latitude longitude and altitude ?
